src/api/app/models/commit_activity.rb

```diff
@@ -0,0 +1,24 @@
+class CommitActivity < ApplicationRecord
+  belongs_to :user
+
+  validates :user, :date, :project, :package, :count, presence: true
+
+  # technically we only want entries with at least count 1, but 0s live
+  # for a brief moment during creation of the activity line
+  validates :count, numericality: { more_than_or_equal_to: 0,
+                                    only_integer: true }
+
+  def self.create_from_event_payload(payload)
+    user = User.find_by_login(payload['user'])
+    params = { user: user, date: Time.zone.today,
+               project: payload['project'], package: payload['package'] }
+    # TODO: Rails 6 supports create_or_find which is race condition free
```

Comment:
[find_or_create_by](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_create_by) is not an option?

-> ` CommitActivity.find_or_create_by(params).increment!(:count) # Assuming that default value of count is 0`